modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/AbstractIndexScan.java

```diff
@@ -132,15 +132,15 @@ public String indexName() {
             cost = 0;
 
             if (lowerCondition() != null) {
-                double selectivity0 = mq.getSelectivity(this, RexUtil.composeDisjunction(builder, lowerCondition()));
+                double selectivity0 = mq.getSelectivity(this, RexUtil.composeConjunction(builder, lowerCondition()));
 
                 selectivity -= 1 - selectivity0;
 
                 cost += Math.log(rows);
             }
 
-            if (upperCondition() != null) {
-                double selectivity0 = mq.getSelectivity(this, RexUtil.composeDisjunction(builder, upperCondition()));
+            if (upperCondition() != null && lowerCondition() != null && !lowerCondition().equals(upperCondition())) {
```

Comment:
> !lowerCondition().equals(upperCondition())
this doesn't work with condition, only with bounds. I think it's be better to leave it as is for now and fix it under https://issues.apache.org/jira/browse/IGNITE-14034

Reply:
discussed in private, i don`t touch this place.